virtcontainers/api.go

```diff
@@ -758,6 +758,46 @@ func StatsContainer(ctx context.Context, sandboxID, containerID string) (Contain
 	return s.StatsContainer(containerID)
 }
 
+// StatsSandbox is the virtcontainers sandox stats entry point.
+// StatsSandbox returns a detailed sandbox stats.
+func StatsSandbox(ctx context.Context, sandboxID string) (SandboxStats, []ContainerStats, error) {
+	span, ctx := trace(ctx, "StatsSandbox")
+	defer span.Finish()
+
+	if sandboxID == "" {
+		return SandboxStats{}, []ContainerStats{}, vcTypes.ErrNeedSandboxID
+	}
+
+	lockFile, err := rLockSandbox(ctx, sandboxID)
+	if err != nil {
+		return SandboxStats{}, []ContainerStats{}, err
+	}
+
+	defer unlockSandbox(ctx, sandboxID, lockFile)
+
+	s, err := fetchSandbox(ctx, sandboxID)
+	if err != nil {
+		return SandboxStats{}, []ContainerStats{}, err
+	}
+	defer s.releaseStatelessSandbox()
+
+	sandboxStats, err := s.Stats()
+	if err != nil {
+		return SandboxStats{}, []ContainerStats{}, err
+	}
+
+	containerStats := []ContainerStats{}
+	for _, c := range s.containers {
+		cstats, err := s.StatsContainer(c.id)
+		if err != nil {
+			return SandboxStats{}, []ContainerStats{}, err
+		}
+		containerStats = append(containerStats, cstats)
+	}
```

Comment:
Is there much utility in returning each container stat set individually?
If a user wanted this, they could use the existing API?
If this is desired, should we consider returning the some of container stats?  WDYT?

Reply:
I'm okay with it as is. We can always sum on the usage side (CLI), and that way we don't lose any potentially interesting information.